test/cadet/assessments_test.exs

```diff
@@ -0,0 +1,200 @@
+      })
+
+    assert mission.title == "sidequest"
+    assert mission.category == :sidequest
+  end
+
+  test "create contest" do
+    {:ok, mission} =
+      Assessments.create_mission(%{
+        title: "contest",
+        category: :contest,
+        open_at: Timex.now(),
+        close_at: Timex.shift(Timex.now(), days: 7)
+      })
+
+    assert mission.title == "contest"
+    assert mission.category == :contest
+  end
+
+  test "create path" do
+    {:ok, mission} =
+      Assessments.create_mission(%{
+        title: "path",
+        category: :path,
+        open_at: Timex.now(),
+        close_at: Timex.shift(Timex.now(), days: 7)
+      })
+
+    assert mission.title == "path"
+    assert mission.category == :path
+  end
+
+  test "create programming question" do
+    mission = insert(:mission)
+
+    {:ok, question} =
+      Assessments.create_question(
+        %{
+          title: "question",
+          weight: 5,
+          type: :programming,
+          question: %{},
+          raw_question: "{\"content\": \"asd\", \"solution_template\": \"template\",
+            \"solution\": \"soln\", \"library\": {\"version\": 1}}"
+        },
+        mission.id
+      )
+
+    assert question.title == "question"
+    assert question.weight == 5
+    assert question.type == :programming
+  end
+
+  test "create multiple choice question" do
+    mission = insert(:mission)
+
+    {:ok, question} =
+      Assessments.create_question(
+        %{
+          title: "question",
+          weight: 5,
+          type: :multiple_choice,
+          question: %{},
+          raw_question:
+            "{\"content\":\"asd\",\"choices\":[{\"is_correct\":true,\"content\":\"asd\"}]}"
+        },
+        mission.id
+      )
+
+    assert question.title == "question"
+    assert question.weight == 5
+    assert question.type == :multiple_choice
+  end
+
+  test "publish mission" do
+    {:ok, mission} =
+      Assessments.create_mission(%{
+        title: "mission",
+        category: :mission,
+        open_at: Timex.now(),
+        close_at: Timex.shift(Timex.now(), days: 7)
+      })
+
+    {:ok, mission} = Assessments.publish_mission(mission.id)
+    assert mission.is_published == true
+  end
+
+  test "update mission" do
+    {:ok, mission} =
+      Assessments.create_mission(%{
+        title: "mission",
+        category: :mission,
+        open_at: Timex.now(),
+        close_at: Timex.shift(Timex.now(), days: 7)
+      })
+
+    Assessments.update_mission(mission.id, %{title: "changed_mission"})
+    mission = Assessments.get_mission(mission.id)
+    assert mission.title == "changed_mission"
+  end
+
+  test "all missions with category" do
+    mission = insert(:mission, category: :mission)
+    sidequest = insert(:mission, category: :sidequest)
+    contest = insert(:mission, category: :contest)
+    path = insert(:mission, category: :path)
+    assert mission.id in Enum.map(Assessments.all_missions(:mission), fn m -> m.id end)
```

Comment:
You can simplify this test. Since you're only adding one for each category, you could simply do `assert [mission] = Assessments.all_missions(:mission)`. Similarly for the other `category`s.